aws/resource_aws_media_store_container.go

```diff
@@ -0,0 +1,149 @@
+package aws
+
+import (
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/mediastore"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsMediaStoreContainer() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsMediaStoreContainerCreate,
+		Read:   resourceAwsMediaStoreContainerRead,
+		Update: resourceAwsMediaStoreContainerUpdate,
+		Delete: resourceAwsMediaStoreContainerDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
+					value := v.(string)
+					if !regexp.MustCompile("^\\w+$").MatchString(value) {
+						errors = append(errors, fmt.Errorf("%q must match \\w+", k))
+					}
+					return
+				},
+			},
+			"policy": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				ValidateFunc:     validateIAMPolicyJson,
+				DiffSuppressFunc: suppressEquivalentAwsPolicyDiffs,
+			},
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"endpoint": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsMediaStoreContainerCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).mediastoreconn
+
+	input := &mediastore.CreateContainerInput{
+		ContainerName: aws.String(d.Get("name").(string)),
+	}
+
+	_, err := conn.CreateContainer(input)
+	if err != nil {
+		return err
+	}
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{mediastore.ContainerStatusCreating},
+		Target:     []string{mediastore.ContainerStatusActive},
+		Refresh:    mediaStoreContainerRefreshStatusFunc(conn, d.Get("name").(string)),
+		Timeout:    10 * time.Minute,
+		Delay:      10 * time.Second,
+		MinTimeout: 3 * time.Second,
+	}
+
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		return fmt.Errorf("[WARN] Error waiting for MediaStore Container status to be \"ACTIVE\": %s", err)
```

Comment:
Nitpick, but I think [the error](https://github.com/hashicorp/terraform/blob/master/helper/resource/error.go#L53) which comes out of `WaitForState` is sufficiently detailed so we don't need to duplicate the context here.